modules/top-level/plugins/default.nix

```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (import ./utils.nix lib)
+    normalizedPluginType
+    normalizePluginList
+    ;
+in
+{
+  options = {
+    finalPlugins = lib.mkOption {
+      visible = false;
+      internal = true;
```

Comment:
Maybe we also want this to be `readOnly = true`?

It's probably _fine_ for users to override this if they want some custom impl, but since it's intended as internal and as part of the implementation of other options, it _feels_ like it should be read-only.